### PR TITLE
HAL builder and Dashboard bugs fixing

### PR DESCRIPTION
**Pull request by marco-scoppetta**
- EngineClient.js is now a static module that does not need instantiation
- fixed useReasoner toggle to work also in Safari
- fixed bug on altClick on generatedRelations
- hal builder more capable of connecting couple of variables in not trivial match queries

### CODE REVIEW

File: grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java [hunk 1]
```diff
@@ -68,74 +71,97 @@ public static String renderHALConceptOntology(Concept concept, String keyspace)
...
+                                }
+                            });
+
                         }
                     });
         }
     }
 
-    private static Map<String, Collection<String>> computeLinkedNodesFromQuery(MatchQuery matchQuery, Map<String, Collection<String>> linkedNodes, Map<String, String> roleTypes) {
+    private static void attachSingleFakeRelation(Representation currentHal, Concept currentVar, Concept otherVar,Map<String, String> roleTypes, String currentVarName, String otherVarName,String relationType) {
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java]: This method scares me. Why do we need a fake relation ?

File: grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java [hunk 2]
```diff
@@ -41,23 +43,24 @@
...
+    private final static String ASSERTION_URL = REST.WebPath.GRAPH_MATCH_QUERY_URI + "?query=match $x id '%s'; $y id '%s'; $r (%s$x, %s$y) %s; select $r;";
     private final static String HAS_ROLE_EDGE = "EMPTY-GRAKN-ROLE";
 
     public static Json renderHALArrayData(MatchQuery matchQuery, Collection<Map<String, Concept>> graqlResultsList, String keyspace) {
+
         //Compute connections between variables in Graql result
-        Map<String, Collection<String>> linkedNodes = new HashMap<>();
-        Map<String, String> roleTypes = new HashMap<>();
+        Map<String, Collection<VarAdmin>> linkedNodes = new HashMap<>();
+        Map<String,Map<String, String>> roleTypes = new HashMap<>();
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java]: I am guessing `roleTypes` and `linkedNodes` are filled in `computeLinkedNodesFromQuery` ? That feels odd I would say it should be something like: `roleTypes = compute....` and `linkedNodes = compute....`. I.e. it should be set through the return type ideally. This is really nit picky though.

Comment (marco-scoppetta) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java]: fffuuuuuuu this is the wrong version of the file! please dont merge just yet

File: grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java (hunk 1 above)
Comment (marco-scoppetta) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java]: wrong version again, wft happeneddd

**Pull request merged**